pennylane/ops/qubit/observables.py

```diff
@@ -438,11 +438,16 @@ class BasisStateProjector(Projector, Operation):
     # The call signature should be the same as Projector.__new__ for the positional
     # arguments, but with free key word arguments.
     def __init__(self, state, wires, id=None):
-        wires = Wires(wires)
-        state = list(qml.math.toarray(state).astype(int))
+        wires = qml.wires.Wires(wires)
 
-        if not set(state).issubset({0, 1}):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {state}")
+        if not qml.math.is_abstract(state):
+            # for cases like [jax.numpy.array(0), jax.numpy.array(1)]
+            state_array = qml.math.asarray(state)
+            state_array = state_array.astype(int)
```

Comment:
```suggestion
            state_array = qml.math.cast(state_array, int)
```

Might potentially work a little bit better.